### ISSUE TITLE
Attribution & Giphy Guidelines

### ISSUE THREAD

**Issue by antoinejaussoin** [sergiop/react-giphy-searchbox#36]
Hi!

Thanks for your amazing component, which I'm using on retrospected.com.

I've just received an email from Giphy, stating that the component doesn't comply to the Attribution Policy:

<img width="284" alt="Spark_-_antoine_jaussoin_com" src="https://user-images.githubusercontent.com/110254/93360293-69cca680-f83b-11ea-9b66-86b69942c0b1.png">

Looking at their website (https://developers.giphy.com/docs/sdk/#design-guidelines), it looks like the component needs to somehow display the image creator's handle/username and perhaps a link to see more from them:

<img width="449" alt="GIPHY_Developers" src="https://user-images.githubusercontent.com/110254/93360254-5de0e480-f83b-11ea-8260-0c700d0e2f0c.png">

My question is: is that something you are planning to work on at some point?

Thanks!

**Comment by sergiop** [comment]
Hi again @antoinejaussoin and thanks for reporting 🙂 .

At the time I developed this module they haven't released this guidelines yet, but I think we can add what they're asking. Do you think it could be enough to add the name of the contributor for each result (linked with the original media on Ghiphy's website), or do you believe we should implement a more complex solution, like the one they are showing in their docs? I understand the needing of give the right attribution to the authors, but I don't want to add too much complexity to the UI, keeping it simple and focused in with the main purpose.

**Comment by sergiop** [comment]
@antoinejaussoin I quickly drafted a possible solution (take a look at the first two images), but I don't like it so much. Adding the name over the image, in our case, significantly covers the image itself. Otherwise, showing it on mouse over, forces us to manage differences between non-touch and touch devices. Any suggestions or feedback?

![with-author](https://user-images.githubusercontent.com/2235134/93758397-a8ca7580-fc08-11ea-84ea-df2e960a7f24.png)

**Comment by sergiop** [comment]
A further thought: opening their website at the [trending page](https://giphy.com/trending-gifs), I can't see any attribution to authors showed over the images on mobile/touch view, neither on `onMouseOver` action. Otherwise on desktop view I can see the the attribution only on `onMouseOver` action. The problem is that our "widget", as well on desktop wiew, take up much less space than their desktop layout.

Anyway... following what they are doing in their website, probably a compromise solution could be to hide the attribution on mobile/touch view, and show it on `onMouseOver` on desktop view.

![Schermata 2020-09-21 alle 14 44 15](https://user-images.githubusercontent.com/2235134/93768158-488fff80-fc19-11ea-8fb1-c57c5442c21f.png)

![Schermata 2020-09-21 alle 14 45 16](https://user-images.githubusercontent.com/2235134/93768179-504fa400-fc19-11ea-8c98-b71f20cccd28.png)

**Comment by antoinejaussoin** [comment]
Hi! Apologies for the late reply.

I was wondering if a simple tooltip over the image wouldn't be enough? As you said, it would only work on desktop but it's probably fine?

**Comment by sergiop** [comment]
@antoinejaussoin I understand the purpose of giving the proper attribution asked by Giphy's team, but this module is intended to have a quick search and select tool, I don't want the user can be distracted by unuseful information. I think that the idea I proposed could fit the main need (as well the tooltip, it’s just an aesthetic issue). I can add it, maybe leaving it optional with a prop, but we have to figure out if the Giphy's dev team will like the way we will add this feature.

**Comment by caleb15** [comment]
I could be wrong but I think you just need a updated logo?

https://developers.giphy.com/faq/

> Glad you asked! As per our section 5A of our Terms of Service, we require all apps that use the GIPHY API to conspicuously display "Powered By GIPHY" attribution marks where the API is utilized. You can find approved official logo marks here. You may also be asked to provide screenshots or a video of app during the application process. 

https://support.giphy.com/hc/en-us/articles/360028134111-GIPHY-API-Terms-of-Service-

Approved logos: https://media.giphy.com/giphy-attribution-marks.zip

**Comment by sergiop** [comment]
> I could be wrong but I think you just need a updated logo?

I don't think so @caleb15, I'm using the `Poweredby_640px-White_HorizText.png` logo you can find inside the approved logos ZIP set. Anyway, with the `poweredByGiphyImage` prop you can change it with the one you prefer from that set. 😉

<img width="1162" alt="Schermata 2020-10-22 alle 10 44 02" src="https://user-images.githubusercontent.com/2235134/96848042-31c90c80-1454-11eb-846c-44f219fea9c6.png">
